### PR TITLE
Don't keep created 0byte file on error

### PR DESCRIPTION
**Pull request by kenhys**

### CODE REVIEW

File: lib/io.c [hunk 1]
```diff
@@ -114,6 +114,7 @@ inline static grn_rc grn_pread(grn_ctx *ctx, fileinfo *fi, void *buf,
                                size_t count, off_t offset);
 inline static grn_rc grn_pwrite(grn_ctx *ctx, fileinfo *fi, void *buf,
                                 size_t count, off_t offset);
+inline static grn_bool grn_path_has_existed(const char *path);
```
Comment (kou) [inline lib/io.c]: Why `has_existed`?
Is `grn_path_exist` enough?

File: lib/io.c [hunk 2]
```diff
@@ -1224,8 +1225,20 @@ grn_io_seg_map_(grn_ctx *ctx, grn_io *io, uint32_t segno, grn_io_mapinfo *info)
     if (!grn_fileinfo_opened(fi)) {
       char path[PATH_MAX];
       gen_pathname(io->path, path, fno);
+      grn_bool path_exists = GRN_TRUE;
```
Comment (kou) [inline lib/io.c]: We don't use "third person singular" form for variable name.
You can use `path_exist`.

Comment (kou) [reply lib/io.c]: We define all variables at the top of block as I mentioned before.

File: lib/io.c [hunk 3]
```diff
@@ -1224,8 +1225,20 @@ grn_io_seg_map_(grn_ctx *ctx, grn_io *io, uint32_t segno, grn_io_mapinfo *info)
...
+      if (!grn_path_has_existed(path)) {
+        path_exists = GRN_FALSE;
+      }
```
Comment (kou) [inline lib/io.c]: It's redundant.

```c
grn_bool path_exist;

path_exist = grn_path_exist(path);
```

File: lib/io.c [hunk 4]
```diff
@@ -1224,8 +1225,20 @@ grn_io_seg_map_(grn_ctx *ctx, grn_io *io, uint32_t segno, grn_io_mapinfo *info)
...
       if (!grn_fileinfo_open(ctx, fi, path, O_RDWR|O_CREAT)) {
         DO_MAP(io, &info->fmo, fi, pos, segment_size, segno, info->map);
+        if (!info->map && path_exists == GRN_FALSE) {
```
Comment (kou) [inline lib/io.c]: `!path_exist`

File: lib/io.c [hunk 5]
```diff
@@ -1224,8 +1225,20 @@ grn_io_seg_map_(grn_ctx *ctx, grn_io *io, uint32_t segno, grn_io_mapinfo *info)
...
+          if (grn_unlink(path) == 0) {
+            GRN_LOG(ctx, GRN_LOG_INFO,
+                    "[io][map][error] removed 0byte file: <%s>", path);
```
Comment (kou) [inline lib/io.c]: Why does it use `0byte` not `empty`?

File: lib/io.c [hunk 6]
```diff
@@ -1548,6 +1561,20 @@ grn_io_get_disk_usage(grn_ctx *ctx, grn_io *io)
...
 }
 
+grn_bool
+grn_path_has_existed(const char *path)
+{
+  if (!path) {
+    return GRN_FALSE;
+  }
+
+  struct stat statbuf;
```
Comment (kou) [inline lib/io.c]: We define variables at the top of block.

File: lib/io.c [hunk 7]
```diff
@@ -1548,6 +1561,20 @@ grn_io_get_disk_usage(grn_ctx *ctx, grn_io *io)
...
+  if (!stat(path, &statbuf)) {
```
Comment (kou) [inline lib/io.c]: It's redundant.

```c
return stat(path, &statbuf) == 0;
```

File: lib/io.c (hunk 6 above)
Comment (kou) [reply lib/io.c]: We use `_` as word separator for variable name.

**Comment by kenhys** [comment]
I've fixed some feedback points.

* Use `grn_path_exist` utility function.
* Change variable name from `path_exists` to `path_exist`
* Use `!path_exist` conditional expression instead of `path_exist == GRN_FALSE`
* Use `empty` in message instead of `0byte`.

**Pull request merged**